### PR TITLE
Embed external xcframeworks regardless of linking type

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/6100

### Short description 📝

This PR supersedes https://github.com/tuist/tuist/pull/6182 and https://github.com/tuist/tuist/pull/6183. I discussed this issue with @kwridan at SwiftHeroes and we both agreed that the best approach for the missing privacy report is to follow what Apple is doing by embedding xcframeworks. That being said, to minimise the impact this change might have (especially regarding resources), we will be only embedding xcframeworks if they are external.

### How to test the changes locally 🧐

Generating privacy report for `app_with_spm_dependencies` should include `GoogleMobileAds`

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistGraph/Graph/GraphDependency.swift [hunk 1]
```diff
@@ -206,11 +208,11 @@ public enum GraphDependency: Hashable, CustomStringConvertible, Comparable, Coda
         }
     }
 
-    public var isPrecompiledDynamicAndLinkable: Bool {
+    public var isPrecompiledAndEmbeddable: Bool {
         switch self {
         case .macro: return false
         case let .xcframework(xcframework):
-            return xcframework.linking == .dynamic
+            return xcframework.linking == .dynamic || xcframework.isExternal
```
Comment (kwridan) [inline Sources/TuistGraph/Graph/GraphDependency.swift:215]: From what I can see, we need to also restrict by xcframework type framework  (`.framework`)

| XCFramework Type | Result | Note |
| ----- | --- | --- |
| Static Library `.a`| ❌ | Embedding simply copies the `.a` to `/Frameworks` with all the symbols included |
| Dynamic Library `.dylib`| ✅ | Embedding copies the `.dylib` to `/Frameworks` as expected |
| Static Framework `.framework`| ✅ | Embedding copies the `.framework` to `/Frameworks` and strips symbols from framework binary |
| Dynamic Framework `.framework`| ✅ | Embedding copies the `.framework` to `/Frameworks` as expected |

e.g. [`MyStaticLibrary.xcframework`](https://github.com/tuist/tuist/tree/main/fixtures/ios_app_with_xcframeworks/XCFrameworks/MyStaticLibrary) gets processed to a `libMyStaticLibrary.a` which shouldn't be embed. 

It doesn't look like we capture the framework type information in the xcframework metadata - we can base it based on the library extension

```suggestion
            return xcframework.linking == .dynamic || (xcframework.isExternal && xcframework.infoPlist.libraries.first?.path.extension == "framework")
```

or can uplift this to be a property on the metadata

```suggestion
            return xcframework.linking == .dynamic || (xcframework.isExternal && xcframework.type == .framework)
```

**Review by kwridan — APPROVED** [review]
Thanks @fortmarek

File: Sources/TuistGraph/Graph/GraphDependency.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistGraph/Graph/GraphDependency.swift]: Gotcha 👍 I updated the PR with the first suggestion

Comment (apps4everyone) [reply Sources/TuistGraph/Graph/GraphDependency.swift]: please see also the wrong linking for xcframeworks https://github.com/tuist/tuist/pull/6153/files

Comment (apps4everyone) [reply Sources/TuistGraph/Graph/GraphDependency.swift]: maybe xcframework.linking is just wrong

**Pull request merged**